### PR TITLE
Merge android-installable-build.ts and force-circleci-status.ts to share installable build logic

### PR DESCRIPTION
**Pull request by jtreanor**
Currently, we have `android-installable-build.ts` to manage Android installable builds and `force-circleci-status.ts` to update statuses for iOS installable builds. We now want to fully support iOS installable builds. To avoid duplication and hopefully keep things simple, I have merged these into one file.

Now `installable-build.ts` does the following:

- If the webhook it receives is `ci/circleci: Installable Build/Hold`:
    - Mark it as green, to avoid the perpetual orange dot.
    - Comment on the PR to say `You can trigger an installable build for these changes by visiting CircleCI here.`.
- If the webhook it receives is `ci/circleci: Installable Build`:
    - Check if there is a `comment.json` artifact on the build. If there is, comment with the `body` key from that.
    - Otherwise, grab a link to the APK on the build and comment with `You can test the changes on this Pull Request by downloading the APK here.`

The changes may seem large, but it is primarily a refactor and the tests are thorough. As usual, testing this is difficult so I have added thorough unit tests. the risk of merging is low however, and I can monitor it closely when merged. If you're not comfortable with this, I can work on setting up a sandbox of some kind to test this.

### CODE REVIEW

File: org/pr/installable-build.ts [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+  }
+
+  const apkArtifact = artifacts.find(artifact => artifact.path.endsWith(".apk"))
+  if (apkArtifact) {
+    return `You can test the changes on this Pull Request by downloading the APK [here](${apkArtifact.url}).`
+  }
+  return undefined
+}
+
+export default async (status: Status) => {
```
Comment (jtreanor) [inline org/pr/installable-build.ts]: Peril runs the default export. I have extracted most of the logic into functions to hopefully make it easy to follow.

File: org/pr/installable-build.ts [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+import fetch from "node-fetch"
+import * as url from "url";
+import * as path from "path";
+
+const CIRCLECI_TOKEN: string = process.env['CIRCLECI_TOKEN']
+const PERIL_BOT_USER_ID: number = parseInt(process.env['PERIL_BOT_USER_ID'], 10)
+
+// This is a list of the CircleCI statuses to process
+const HOLD_CONTEXTS: string[] = ["ci/circleci: Installable Build/Hold"]
+const INSTALLABLE_BUILD_CONTEXTS: string[] = ["ci/circleci: Installable Build", "ci/circleci: Test Android on Device"]
```
Comment (jtreanor) [inline org/pr/installable-build.ts]: If a repo want to take advantage of this logic with a different job name, it just needs to be added here.

File: org/pr/installable-build.ts [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+  if (res.ok) {
+    return res.json()
+  }
+  return []
+}
+
+async function getDownloadCommentText(status) {
+  const artifacts = await circleCIArtifacts(status)
+
+  const commentJsonArtifact = artifacts.find(artifact => artifact.path.endsWith("comment.json"))
```
Comment (jtreanor) [inline org/pr/installable-build.ts]: Getting the repos themselves to define the comment in a `comment.json` file allows this logic to work across platforms. e.g. iOS can mention HockeyApp and Android can mention downloading an APK.

File: org/pr/installable-build.ts [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+  if (commentJsonArtifact) {
+    // Download the JSON file so we can get the comment text
+    const res = await fetch(commentJsonArtifact.url)
+    if (res.ok) {
+      return res.json()["body"]
+    }
+  }
+
+  const apkArtifact = artifacts.find(artifact => artifact.path.endsWith(".apk"))
```
Comment (jtreanor) [inline org/pr/installable-build.ts]: Even though `comment.json` gives more flexibility, keeping the existing APK logic here means everything continues to work as expected.

File: org/pr/installable-build.ts (hunk 2 above)
Comment (jkmassel) [reply org/pr/installable-build.ts]: This is really nice :)

**Pull request merged**